flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/conversion/HiveInspectors.java

```diff
@@ -450,4 +454,56 @@ private static ObjectInspector getObjectInspector(TypeInfo type) {
 	public static DataType toFlinkType(ObjectInspector inspector) {
 		return HiveTypeUtil.toFlinkType(TypeInfoUtils.getTypeInfoFromTypeString(inspector.getTypeName()));
 	}
+
+	// given a Hive ObjectInspector, get the class for corresponding Flink object
+	private static Class<?> classForObjectInspector(ObjectInspector inspector) {
+		switch (inspector.getCategory()) {
+			case PRIMITIVE: {
+				PrimitiveObjectInspector primitiveOI = (PrimitiveObjectInspector) inspector;
+				switch (primitiveOI.getPrimitiveCategory()) {
+					case STRING:
+					case CHAR:
+					case VARCHAR:
+						return String.class;
+					case INT:
+						return Integer.class;
+					case LONG:
+						return Long.class;
+					case BYTE:
+						return Byte.class;
+					case SHORT:
+						return Short.class;
+					case FLOAT:
+						return Float.class;
+					case DOUBLE:
+						return Double.class;
+					case DECIMAL:
+						return BigDecimal.class;
+					case BOOLEAN:
+						return Boolean.class;
+					case BINARY:
+						return byte[].class;
+					case DATE:
+						return Date.class;
+					case TIMESTAMP:
```

Comment:
why don't we support TIMESTAMP?

Reply:
We don't support timestamp due to precision difference between Flink (precision 3) and Hive (precision 9).

Reply:
If so, we may need to disable support for timestamp in all util methods in HiveInspectors, instead of just this one. Can you create a jira to do it?

Reply:
Not sure whether we have to disable it in the util methods. It's a known limitation documented [here](https://ci.apache.org/projects/flink/flink-docs-release-1.9/dev/table/hive/#limitations). So users shouldn't be using it in the first place.
Also remembered we already have a JIRA for the bridged types: https://issues.apache.org/jira/browse/FLINK-13438 and we can continue the work there.

Reply:
cool. minor: Can you add a comment here?

Reply:
let's discuss offline

Reply:
Flink timestamp can have a higher precision than Hive's, why converting Hive timestamp to Flink's doesn't work? Seems we need to be cautious when converting Flink timestamp to Hive's, right?